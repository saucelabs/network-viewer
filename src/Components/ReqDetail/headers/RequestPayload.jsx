import React, { useMemo } from 'react';
import PropTypes from 'prop-types';

import Styles from './../Headers.styles.scss';
import { parseRequestPayload } from '../../../utils';

const RequestPayload = ({ data, isPayloadTransformed }) => {
  const payloadData = data.headers.postData.text;
  const parsedData = useMemo(() => parseRequestPayload(payloadData), [payloadData]);

  return (
    <div className={Styles['header-detail']}>
      {isPayloadTransformed ? (
        <div className={Styles['response-content']}>
          <pre className={Styles['log-body-colorless']}>
            {parsedData}
          </pre>
        </div>
      ) : payloadData}
    </div>
  );
};

RequestPayload.propTypes = {
  data: PropTypes.object,
  isPayloadTransformed: PropTypes.bool.isRequired,
};

RequestPayload.defaultProps = {
  data: null,
};

export default RequestPayload;
