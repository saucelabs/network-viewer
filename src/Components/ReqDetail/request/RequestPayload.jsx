import React, { useMemo } from 'react';
import PropTypes from 'prop-types';

import Styles from '../ReqDetail.styles.scss';
import { parseRequestPayload } from '../../../utils';
import CopyAllButton from '../CopyAllButton';

const RequestPayload = ({ data, isPayloadTransformed }) => {
  const payloadData = data.headers.postData.text;
  const parsedData = useMemo(() => parseRequestPayload(payloadData), [payloadData]);
  const payload = isPayloadTransformed ? parsedData : payloadData;

  return (
    <div className={Styles['section-detail']}>
      <div className={Styles['payload-content']}>
        <div className={Styles['copy-button']}>
          <CopyAllButton text={payload} />
        </div>
        <div className={Styles['payload-body']}>
          {payload}
        </div>
      </div>
    </div>
  );
};

RequestPayload.propTypes = {
  data: PropTypes.object,
  isPayloadTransformed: PropTypes.bool.isRequired,
};

RequestPayload.defaultProps = {
  data: null,
};

export default RequestPayload;
