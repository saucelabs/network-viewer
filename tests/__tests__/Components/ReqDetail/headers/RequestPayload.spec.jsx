import React from 'react';
import { shallow } from 'enzyme';

import RequestPayload from '../../../../../src/Components/ReqDetail/request/RequestPayload';

describe('RequestPayload', () => {
  const props = {
    data: {
      headers: {
        postData: {
          text: "{ 'name': 'foo' }",
        },
      },
    },
    isPayloadTransformed: true,
  };

  it('renders without crashing', () => {
    const element = shallow(<RequestPayload {...props} />);
    expect(element).toMatchSnapshot();
  });
});
