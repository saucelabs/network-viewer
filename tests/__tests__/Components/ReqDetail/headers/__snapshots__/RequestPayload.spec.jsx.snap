// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequestPayload renders without crashing 1`] = `
<div>
  <div>
    <div>
      <CopyAllButton
        text="{ 'name': 'foo' }"
      />
    </div>
    <div>
      { 'name': 'foo' }
    </div>
  </div>
</div>
`;
